### PR TITLE
[Ink] Add fail the ripple gesture when other scroll view gestures are competing unless they are tap or long press gestures to MDCInkTouchController

### PR DESCRIPTION
**Pull request by maicki**
Mirroring #6487

### CODE REVIEW

**Review by jverkoey — APPROVED** [review]

File: components/Ink/src/MDCInkTouchController.h [hunk 1]
```diff
@@ -57,6 +57,14 @@
  */
 @property(nonatomic, assign) BOOL cancelsOnDragOut;
 
+/**
+ If enabled, the ink gesture will require failure of UIScrollView gesture recognizers in order to
+ activate.
+
+ Defaults to NO.
+ */
+@property(nonatomic, assign) BOOL requiresFailureOfScrollViewGestures;
```
Comment (yarneo) [inline components/Ink/src/MDCInkTouchController.h:66]: I don't think the new naming and documentation is clear for me as a client.

When it is set to YES it doesn't require failure of UIScrollView gesture recognizers. The UIScrollView gesture recognizers aren't the ones who are failing, but rather the MDCInkGestureRecognizer is failing if there are any gestures of the UIScrollView that aren't taps or long presses.

**Review by yarneo — CHANGES_REQUESTED** [review]

File: components/Ink/src/MDCInkTouchController.h (hunk 1 above)
Comment (yarneo) [reply components/Ink/src/MDCInkTouchController.h:66]: @jverkoey ^

Comment (jverkoey) [reply components/Ink/src/MDCInkTouchController.h:66]: That's the opposite of what the method does.

When this property is enabled, `shouldRequireFailureOfGestureRecognizer: otherGestureRecognizer ` will be able to return YES, indicating that `otherGestureRecognizer ` should fail prior to `gestureRecognizer` being able to enter `UIGestureRecognizerStateBegan`.

Comment (yarneo) [reply components/Ink/src/MDCInkTouchController.h:66]: ah sorry, my bad here. I thought we were talking about `gestureRecognizer:shouldBeRequiredToFailByGestureRecognizer:`. 
Thanks for clarifying

**Review by yarneo — APPROVED** [review]

File: components/Ink/src/MDCInkTouchController.h (hunk 1 above)
Comment (jverkoey) [reply components/Ink/src/MDCInkTouchController.h:66]: All good! The gesture recognizer state machine's got a good amount going on.

**Pull request merged**